core/commands/diff.py

```diff
@@ -174,51 +178,165 @@ def run(self, edit, cursors=None, navigate_to_next_hunk=False):
                 return
             raise err
 
-        text = prelude + '\n--\n' + stdout
+        old_diff = self.view.settings().get("git_savvy.diff_view.raw_diff")
+        self.view.settings().set("git_savvy.diff_view.raw_diff", diff)
+        text = prelude + '\n--\n' + diff
 
         self.view.run_command(
             "gs_replace_view_text", {"text": text, "restore_cursors": True}
         )
-        if navigate_to_next_hunk:
+        if not old_diff:
             self.view.run_command("gs_diff_navigate")
 
 
 class GsDiffToggleSetting(TextCommand):
 
     """
-    Toggle view settings: `ignore_whitespace` , `show_word_diff` or
-    `in_cached_mode`.
+    Toggle view settings: `ignore_whitespace` , or `show_word_diff`.
     """
 
     def run(self, edit, setting):
+        settings = self.view.settings()
+
+        setting_str = "git_savvy.diff_view.{}".format(setting)
+        current_mode = settings.get(setting_str)
+        next_mode = not current_mode
+        settings.set(setting_str, next_mode)
+        self.view.window().status_message("{} is now {}".format(setting, next_mode))
+
+        self.view.run_command("gs_diff_refresh")
+
+
+class GsDiffToggleCachedMode(TextCommand):
+
+    """
+    Toggle `in_cached_mode`.
+    """
+
+    # NOTE: MUST NOT be async, otherwise `view.show` will not update the view 100%!
+    def run(self, edit):
+        setting = 'in_cached_mode'
+
         if (
-            setting == 'in_cached_mode'
-            and self.view.settings().get("git_savvy.diff_view.base_commit")
+            self.view.settings().get("git_savvy.diff_view.base_commit")
             and self.view.settings().get("git_savvy.diff_view.target_commit")
         ):
             # There is no cached mode if you diff between two commits, so
             # we need to abort here
```

Comment:
No need to forbid this here, we can exchange `base` with `target` making it essentially a `-R`everse diff.